### ISSUE TITLE
Convergence Issues with Unbalanced 123-Node Test Feeder

### ISSUE THREAD

**Issue by brettross** [e2nIEE/pandapower#889]
Hello,

I have been testing the newly released run3pp functionality against a model I have of the unbalanced IEEE 123-node test feeder. I have been experiencing convergence issues. After a single iteration of NR, the solution converges to a s_mismatch value in the range of 0.19 - 0.5 MW. I do not know if this is a limitation of the solution method or if there is a bug in the problem formulation.

Some information:
- Single phase and two phase lines are approximated with Z1/Z0 values
- Regulator controls are disabled for troubleshooting and taps are locked at neutral. No voltage instabilities observed in solutions
- Single phase shunts are approximated with asymmetric loads
- With the system operated radially, the solution converges to a 0.5 MW maximum error in one iteration. With all switches closed, the network can be operated meshed, though there are still many lateral branches. This reduces the error to 0.2 MW. 
- Making load power balanced or zero does not reduce the maximum error below 0.2 MW
- The 123-node model converges to a small error with runpp. This was done both with balanced loads and by running three powerflows in a three-phase decoupled approximation (e.g., transposed lines, Z1 = Z0)

Is it expected that the current injection method for I2/I0 will have such issues when the unbalance is above some percentage? Or is this result surprising?

Please let me know if you need additional information.

Thanks in advance for your help,
Brett

**Comment by lthurner** [comment]
> Is it expected that the current injection method for I2/I0 will have such issues when the unbalance is above some percentage? Or is this result surprising?

Its definitely not expected. We tested some cases with pretty high unbalance and didn't encounter any issues. However, I am not surprised that the 3 phase power flow has some teething problems. Just from the description it is difficult to see what is going wrong though. Is it possible for you to provide the test network?

**Comment by brettross** [comment]
That is understandable--I will reach out to you via e-mail and provide some more details. Thanks for the help.

**Comment by brettross** [comment]
While I was getting results together I discovered the root of my issue. 

I was neglecting to define all the impedance related properties for my external grid object. I didn't realize that the impedance values were being used in power flow (I was under the impression they were just for short-circuit calculations). Consequently I was making a zero impedance branch on my external grid. This of course gave the solver issues. I have specified a small but nonzero impedance and my convergence issues have disappeared.

Apologies for the misinformation, and thank you for your time.
